Functional.Primitives/ResultFactory.cs

```diff
@@ -167,11 +173,11 @@ public static class Result
 			{
 				successFactory.Invoke();
 
-				return Success<Unit, Exception>(Functional.Unit.Value);
+				return Success<Unit, TException>(Functional.Unit.Value);
 			}
-			catch (Exception ex)
+			catch (TException ex)
```

Comment:
This still leaves room for an Exception that is different than TException to be thrown out of this try. Is there any value to return an Option<TException> in case of failure so that case is explicitly handled?
Could also return a Union<TException, Exception> but for some reason I don't like that as much.

Reply:
Yep, that's the use case I'm looking to address here: only catch a specific exception, and let other exceptions be thrown.

Reply:
Should have read the conversation you guys had above.
Cool cool.

Reply:
Consider that Result is used to express errors, but not every exception is an error (which is why, in general, catching `Exception` and always turning it into an error can be an anti-pattern).
 
.NET not allowing you to catch a `StackOverflowException` is a concrete example of the CLR forcing this perspective on you.